test/blackbox-tests/test-cases/pkg/opam-package-with-subst.t

```diff
@@ -0,0 +1,54 @@
+We test how opam files with substs fields are translated into the dune.lock file.
+ 
+  $ . ./helpers.sh
+
+Generate a mock opam repository
+  $ mkdir -p mock-opam-repository
+  $ cat >mock-opam-repository/repo <<EOF
+  > opam-version: "2.0"
+  > EOF
+
+Make a package with a substs field 
+  $ mkpkg with-substs <<EOF
+  > opam-version: "2.0"
+  > substs: ["foo.ml"]
+  > build: [ ["[" "-e" "foo.ml" "]"] [ "cat" "foo.ml"] ]
```

Comment:
How about just `"-e foo.ml && cat foo.ml"`?

Reply:
Sure let me try that.

Reply:
Well it will have to be `[ -e foo.ml ] && cat foo.ml` however since run is being used that doesn't work. Let me also submit a fix for that.